### ISSUE TITLE
Use logo compatible with both light and dark modes

### ISSUE THREAD

**Issue by fbricon** [redhat-developer/vscode-quarkus#96]
Currently, the extension is published to the marketplace with a logo compatible with a dark mode:
On the extension page of the marketplace:
<img width="580" alt="Screen Shot 2019-09-27 at 4 07 36 PM" src="https://user-images.githubusercontent.com/148698/65775745-0c7aa980-e141-11e9-9966-28a6ccfb815e.png">

In vscode's extensions view :
<img width="308" alt="Screen Shot 2019-09-27 at 4 08 55 PM" src="https://user-images.githubusercontent.com/148698/65775793-22886a00-e141-11e9-8430-fe9f66ff0e7d.png">

But it doesn't work in the publisher page, when displayed in a card:
<img width="193" alt="Screen Shot 2019-09-27 at 4 07 53 PM" src="https://user-images.githubusercontent.com/148698/65775734-07b5f580-e141-11e9-8869-afe35dd3b117.png">

It was proposed to replace the logo with a version compatible with a light mode:
![](https://design.jboss.org/quarkus/logo/final/PNG/quarkus_icon_rgb_128px_default.png)
That should work with the extension page on the marketplace and the publisher card, but not when viewed from within vscode, basically amounting to:

<img width="140" alt="Screen Shot 2019-09-27 at 4 13 06 PM" src="https://user-images.githubusercontent.com/148698/65776144-c7a34280-e141-11e9-8a63-f11611c70fc3.png">

I think the best solution is to get logos with non-transparent cube edges:
- white edges for dark cube, 
- black edges for white cube

cc @maxandersen

**Comment by fbricon** [comment]
Maybe @insectengine can help ;-)

**Comment by insectengine** [comment]
Hello @fbricon, I forwarded my original email I sent Max regarding this, but here is the content of the email I sent...

TLDR is we use the dark cubed version of the logo and a custom color for the background.

______

As per Max, the VSCode Marketplace uses a single image for the Card on the main page and the detail page.

Main marketplace: https://marketplace.visualstudio.com/search?term=quarkus&target=VSCode&category=All%20categories&sortBy=Relevance
Detail page: https://marketplace.visualstudio.com/items?itemName=redhat.vscode-quarkus

The issue is the card has a light background while the detail page has been set to a dark background. We’re currently using the logo designed for a dark background and parts are disappearing on the card view.


Looking at other pages, it seems we can control the background color of the detail page (so it’s not the default light grey).

Here are a couple examples of what other teams have done.
https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-docker
https://marketplace.visualstudio.com/items?itemName=HookyQR.beautify
https://marketplace.visualstudio.com/items?itemName=ms-edgedevtools.vscode-edge-devtools

I propose this solution:
https://design.jboss.org/quarkus/vscodeicon/images/vscodemarketplace_quarkusmarkup.png

1) We use the default icon for the card and detail page:
https://design.jboss.org/quarkus/logo/final/PNG/quarkus_icon_rgb_128px_default.png
(It has the dark cube in the middle so you may not see it because of the transparent background in Chrome)

2) We then set the background color of the detail page to a light blue derived from our blue color (#d8ebff)

3) If possible, we change the hyperlink for the “quarkus” title to our Medium Blue (#5995e4)

**Comment by fbricon** [comment]
@insectengine your plan only fixes the marketplace page, but not the extension page *INSIDE* vscode, as I described it, which will get way way more traffic than the red hat publisher page

**Comment by insectengine** [comment]
@fbricon  - ah ok.. can you post a link to the page where it doesn't work so I can check it out?

**Comment by fbricon** [comment]
you need to install vscode (dark theme should be enabled by default), open the extension page (5th icon with squares on the left). Type quarkus to look for the extensions on the marketplace :
<img width="1125" alt="Screen Shot 2019-09-30 at 4 31 15 PM" src="https://user-images.githubusercontent.com/148698/65888433-ccfdc880-e39f-11e9-80d3-a503cf774f7c.png">

If we used the logo you recommended, it'd be black square on dark background

**Comment by insectengine** [comment]
@fbricon  - question for you.. they only allow one logo throughout all of this process? It seems odd that they would have an option for light and dark on their site and only dark inside vs code. If possible, we should use the reverse logo inside the dark environment and the default one with the dark cube for the other pages on the site itself. Is this possible?

**Comment by fbricon** [comment]
Nope. Single icon for both the marketplace website and the extension view within VS Code.
As I said, the simplest solution would be to provide logos with colored edges. They'll work in both dark and light themes.

**Comment by insectengine** [comment]
We don't want to create a new version of the logo... it would ultimately cause more problems down the line.

How does one using the light background look on VS Code?

https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp
or
https://marketplace.visualstudio.com/items?itemName=kirkslota.karate-runner

**Comment by fbricon** [comment]
C#:
<img width="445" alt="Screen Shot 2019-09-30 at 4 48 20 PM" src="https://user-images.githubusercontent.com/148698/65889817-2f57c880-e3a2-11e9-9d5e-5b063766fb3a.png">

Karate Runner:
<img width="446" alt="Screen Shot 2019-09-30 at 4 47 49 PM" src="https://user-images.githubusercontent.com/148698/65889796-2830ba80-e3a2-11e9-8c80-83a8c536deba.png">

**Comment by insectengine** [comment]
@fbricon  - I suggest we take a page from the karate runner usage. Let's put it on a background of the dark quarkus blue and have it mirror the logo's rounded corners.

So we have this icon (128x128):
https://design.jboss.org/quarkus/vscodeicon/images/quarkus_vscode_icon.png

It would look like this on the light card (the first one on the left):
https://design.jboss.org/quarkus/vscodeicon/images/vscodemarketplace_main_quarkusmarkup_v2.png

Then on the detail page:
https://design.jboss.org/quarkus/vscodeicon/images/vscodemarketplace_quarkusmarkup_v2.png

Finally inside VS code on their default dark:
https://design.jboss.org/quarkus/vscodeicon/images/vscode_interface_dark.png

**Comment by fbricon** [comment]
@insectengine fine by me, however it'll take a few days before we publish a new release on the VS Code Marketplace

**Comment by fbricon** [comment]
@xorye please update the icon to https://design.jboss.org/quarkus/vscodeicon/images/quarkus_vscode_icon.png, and in https://github.com/redhat-developer/vscode-quarkus/blob/master/package.json#L15, change the color to `#d8ebff` and the theme to `light`

**Comment by insectengine** [comment]
Awesome... thanks @fbricon !
